code/code.py

```diff
@@ -20,7 +20,10 @@
 api_hash = os.getenv('TELEGRAM_API_HASH')
 api_username = os.getenv('TELEGRAM_USERNAME')
 
-cluster = "mongodb+srv://"+api_username+":"+api_hash+"@cluster0.yon6aco.mongodb.net/smartSpendDB?retryWrites=true&w=majority"
+cluster = os.getenv('MONGO_DB_URL')
```

Comment:
why have you added this? We don't need to get url from env var as we already have it below in code

Reply:
I added this because setting up the online connection all the time is just wasted effort if we are only testing it. We could simply install mongo on our local machine and have it working in no time so I added a similar mechanism for that too while still leaving the original URL so that nobody is affected.